lib/excon.rb

```diff
@@ -42,13 +42,23 @@ class << self
 
     # @return [Hash] defaults for Excon connections
     def defaults
-      @defaults ||= DEFAULTS
+      defaults = Thread.current.thread_variable_get(:excon_defaults)
+      return defaults if defaults
+      Thread.current.thread_variable_set(:excon_defaults, DEFAULTS.dup)
     end
 
     # Change defaults for Excon connections
     # @return [Hash] defaults for Excon connections
     def defaults=(new_defaults)
-      @defaults = new_defaults
+      Thread.current.thread_variable_set(:excon_defaults, new_defaults)
```

Comment:
I kinda would prefer to do `new_defaults.dup` here. Values in it can be mutated later, and currently that would mutate the hash in the caller too.